pulser-core/pulser/backend/qpu.py

```diff
@@ -61,10 +69,51 @@ def run(
                     f"device ({max_runs})" + suffix
                 )
         results = self._connection.submit(
-            self._sequence, job_params=job_params, wait=wait
+            self._sequence,
+            job_params=job_params,
+            wait=wait,
+            submission_id=self.submission_id,
         )
         return cast(RemoteResults, results)
 
+    def open_submission(self, submission_id: str | None = None) -> Self:
+        """
+        If provided a submission_id, we can add new jobs to a batch within
+        the scope of the context manager otherwise an new open submission
+        is created that you can continuously add jobs to.
+
+        Args:
+            submission_id: An optional unique identifier for an already
+                 open submission
+
+        Returns:
+            self reference for current object
+        """
+        if submission_id:
+            self.submission_id = submission_id
+        else:
+            submission = cast(
+                RemoteResults, self._connection.submit(self._sequence)
+            )
+            self.submission_id = submission.submission_id
+        return self
+
+    def __enter__(self) -> Self:
+        # open returns an instance of self to use open_submission within a
+        # context manager
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
+        # exiting the context of an open submission will
+        # make a remote call to close it.
+        if self.submission_id:
+            self.close_submission(self.submission_id)
```

Comment:
Following up on the idea that `close_submission()` should not be exposed as a public method, this should used the connection directly
```suggestion
            self._connection.close_submission(submission_id)
```